src/app/repository/repository-model.ts

```diff
@@ -0,0 +1,82 @@
+import {iCpeListResource, iPage, iProject} from "./cpe-api.service";
+
+/**
+ * The base class for all resources. Used by inheritance as well as for lists.
+ */
+export class CpeResource {
+  readonly _id: string = '';
```

Comment:
I would only use underscores for private class members.

Reply:
Not my code

Reply:
Changed to protected - maybe we will create one or set one ...